rmf_gazebo_plugins/src/teleport.cpp

```diff
@@ -63,41 +64,45 @@ class TeleportPlugin : public gazebo::ModelPlugin
     get_sdf_param_required<Pose3d>(_sdf, "unload_pose", _unload_pose);
 
     _node = gazebo_ros::Node::Get(_sdf);
-    std::cout << "Started teleport_plugin node..." <<std::endl;
+    std::cout << "Started teleport_plugin node..." << std::endl;
 
     _result_sub = _node->create_subscription<DispenserResult>(
-        "/dispenser_results",
-        rclcpp::SystemDefaultsQoS(),
-        [&](DispenserResult::UniquePtr msg)
-        {
-          dispenser_result_cb(std::move(msg));
-        });
-    
+        "/dispenser_results", rclcpp::SystemDefaultsQoS(),
+        [&](DispenserResult::UniquePtr msg) { dispenser_result_cb(std::move(msg)); });
+
+    _initial_pose = _model->WorldPose();
+
     _load_complete = true;
   }
 
   // Called by the world update start event
   void dispenser_result_cb(DispenserResult::UniquePtr msg)
   {
-    auto status = msg->status;
+    auto status             = msg->status;
     std::string source_guid = msg->source_guid.c_str();
 
     if (source_guid == _load_guid && !_object_loaded)
     {
-      RCLCPP_INFO(_node->get_logger(),  "Loading object");
+      RCLCPP_INFO(_node->get_logger(), "Loading object");
       _model->SetWorldPose(_load_pose);
       _object_loaded = true;
     }
     else if (source_guid == _unload_guid && _object_loaded)
     {
-      RCLCPP_INFO(_node->get_logger(),  "Unloading object");
+      RCLCPP_INFO(_node->get_logger(), "Unloading object");
       _model->SetWorldPose(_unload_pose);
+
+      // BH: Hard coded: Leave object at goal location for 3.0 second, then
+      // teleport
+      // it back to initial ( pre pickup  ) location
+      rclcpp::sleep_for(std::chrono::nanoseconds(3000000000));
```

Comment:
Can we only keep these three lines and the definition of `_initial_pose` while undoing the proposed formatting changes in the rest of the files?

Reply:
so sorry, i realized i messed up the formatting as my vim setup does k&r formatting automatically. It might be awesome to use  something like `.clang-format` to help with formatting as well. Let me patch this up